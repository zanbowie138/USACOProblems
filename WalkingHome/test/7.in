50
13 3
.....H.......
.............
.............
.............
.............
.............
.H...........
.............
......H......
.............
..H...H......
H............
.H...........
9 3
.........
........H
.........
.........
.........
..H......
.........
.........
.........
15 3
...............
...............
............H..
..H....H.......
...............
H..............
...............
...............
...............
...............
...............
...............
...............
...............
...............
27 3
...........................
...........................
...........................
...........................
.....H..H............H.....
...............H...........
..................H........
.................H.........
...........................
...........H...........H...
...........................
..HH.......................
......................H...H
...........................
...........................
...........................
...H.......................
......................H....
......................H....
...............H...........
...........................
......H........H...........
...........................
...H.......................
...........................
...................H.......
...........................
10 3
..........
..........
..........
..........
..........
.......H..
..........
.....H....
..........
H.........
4 3
....
..H.
H...
....
50 3
.....................H............................
..................................................
..................................................
..................................................
............................H.....................
...H..............................................
.........H..............H.........................
........H.........................................
..................................................
...............H.....................H............
...................................H..............
..................................................
..................................................
..................................................
.................H................................
..................................................
..................................................
..................................................
..............................H....H..............
........................H.........................
..................................................
.........H........................................
..................................................
..................................................
..................................................
...........H.........H............................
..................................................
..................................................
..................................................
................H.................................
..................................................
......................H...........................
..H...............................................
......................................H...........
..H.....................................H.........
..................................................
............H.....................................
.............................................H....
..............H...................................
..................................................
..................................................
..................................................
..................................................
..................................................
................................................H.
..................................................
..........................................H.......
.............................................H....
..................................................
..................................................
41 3
.........................................
.........................................
.........................................
.........................................
.........................................
.........................................
.........................................
...............H.........................
.......................H..H..............
.............H...........................
.........................................
.........................................
.........................................
.........................................
.........................................
.................H.......................
.........................................
.........................................
.............H...........................
.........................................
.........................................
..................H.......H..............
.........................................
....H...............................H....
.......................................H.
................................H........
.........................................
...H.....................................
.........................................
...........H.............................
.........................................
.........................................
...................H.....................
.........................................
.........................................
.........................................
...........................H.H...........
.........................................
........................................H
.........................................
.................H.......................
22 3
......................
......................
......................
......................
......................
......................
............H.........
.................H....
......................
......................
H.....................
......................
................H.....
......................
................H.....
..........H...........
......................
...............H......
...H..................
...........H..........
.H....................
......................
39 3
.......................................
.................H.....................
.......................................
.......................................
..........H............................
.......................................
.......................................
.......................................
.......................................
.......................................
..................................H....
.......................................
.......................................
.......................................
.......................................
.................................H.....
.......................................
.......................................
.......................................
...................H...................
.......................................
...............H..................H....
.......................................
.............H.........................
.......................................
...........H...........................
...H...................................
.......................................
...........................H...........
.......................................
...................H...................
.......................................
.......................................
.......H...............................
......H................................
......H..............H.................
.......................................
.......................................
.......................................
13 3
.H.H.........
.............
..H..........
..H..........
.............
..H......H...
.............
..H..........
............H
.............
.............
.............
.H.....H.....
15 3
........H......
............H..
...............
...............
..........H....
...............
.......H.......
...............
...............
...............
...............
...............
......H........
...............
...............
5 3
.....
....H
.....
H....
H....
26 3
..........................
..........................
..........................
..........................
..........................
..........................
..........................
..........................
..........................
.........H................
.....H....................
..........................
..........................
......H...H...............
..H.......................
..........................
..H.......................
.....H....................
..........................
..........................
..........................
..........................
..........................
.........................H
..........................
..........................
27 3
...........................
........................H..
...........................
........................H..
..........................H
...........................
........................H..
...........................
...........................
...........................
...........................
...........................
...........................
...........H...............
...........................
...........................
...........................
...........................
...........................
...........................
..................H........
...........................
...........................
...H....H..................
...........................
...........................
...........................
10 3
..........
..........
..........
..........
..........
..........
..........
..........
.........H
..........
35 3
...................................
...................................
...................................
...................................
...................................
...................................
...................................
...................................
...................................
...................................
..............................H....
............................H......
...................................
...................................
...................................
...................................
........................H..........
...................................
...................................
...................................
...................................
...................................
.......H..................H........
...................................
...........................H.......
...........H........H..............
...................................
..............H..............H.....
...................................
...................................
...................................
...................................
...................H...............
...................................
.....................H.............
49 3
............H....................................
.................................................
................H................................
.................................................
.................................................
.................................................
.................................................
.................................................
.................................................
................................H................
..................H..............................
.................................................
.................................................
.................................................
...H.............................................
.................................................
.................................................
............H....................................
.................................................
.................................................
.................................................
.................................................
..............H...........................H....H.
...............................................H.
.................................................
..........................H......................
.................................................
.................................................
H..........................H.....................
.................................................
.................................................
.................................................
.................................................
....................H............................
.................................................
.................................................
.................................................
.................................................
.................................................
....................................H....H.......
.................................................
.................................................
.................................................
................HH...............................
..............................H..................
.................................................
.................................................
.............H.......H...........................
.................................................
27 3
..H.........H..............
...........................
...........................
...........................
...........................
...H.......................
...........................
...........................
...........................
.........H.................
......................HH...
........H..................
...........................
...........................
H..........................
..............H...H........
...........................
...........................
...........................
...........................
...................H.......
...........................
...........H...............
.......H..................H
...........................
...........................
................H..........
6 3
......
......
......
......
.H....
....H.
35 3
...................................
...................................
...................................
...H...............................
.............................H.....
...................................
...................................
...................................
...................................
.......................H...........
...................................
.......................H...........
...................................
...................................
................H..................
...................................
...................................
...................................
..........................H........
.................H.................
...H...............................
..............H..H...H.............
...................H...............
.......................H...........
...................................
......H............................
...................................
...................................
..............H....................
...................................
...................................
...................................
...................................
...................................
...................................
10 3
..H.......
.....H....
.....H....
..........
..........
......H...
..........
..........
..........
..........
7 3
...H...
.......
.......
......H
.......
.......
...HH..
24 3
..H.....................
........................
H.......................
........................
........................
........................
........H...............
H....................H..
........................
........................
........................
........................
........................
............H...........
........................
........................
........................
........................
........................
........................
............H...........
........H...............
........................
........................
23 3
......H................
.......................
.......................
.......................
.......................
.H..................H..
.......................
........H..............
.......................
........H..............
..............H........
.......................
.......................
.............H.......H.
.......................
..............H........
.......................
................H......
.......................
..............H........
.......................
.......................
.......................
40 3
..H.....................................
...............................H........
...................H........H...........
........................................
........................................
................................H.......
.H......................................
.................................H......
........................................
........................................
........................................
........................................
.....................................H..
........................................
........................................
........................................
..................................H.....
.........H..............................
........................................
................H....H..................
.......................H................
......H.................................
..................................H.....
........................................
........................................
.........................H..............
........................................
........................................
........................................
........................................
.................H......................
........................................
........................................
........................................
..........H.....................H...H...
...H....................................
........................................
........................................
.................H......................
........................................
42 3
......................................H...
..........................................
..........................................
...................................H.H....
H.........................................
..........................................
.........................................H
..............................H...........
H.........................................
..........................................
...........................H..............
..........................................
....................H.....................
..........................................
..........................................
..........................................
..........................................
..........................................
..........................................
..........................................
..........................................
..........................................
.....................................H....
..........................................
..........................................
..........................................
..........................................
..........................................
..H...............H.......................
..........................................
..........................................
....H.....................................
..........................................
...............................H..........
..............H...........................
......H...................................
..........................................
........................................H.
......H...................................
................H.........................
......H...................................
..........................................
15 3
...............
............H..
...............
...............
...............
..H...H........
...............
...............
..........H....
...............
...............
...............
.......H.......
......H........
...............
13 3
.............
.............
....H........
.............
.............
.H...........
.............
.............
.............
...H.........
.............
H.......H....
.............
37 3
.....................................
.........................H...........
.....................................
.....................HH..............
.....................................
.....................................
...............H.....................
.........H...........................
............................H........
..........................H..........
.....................................
......................H...........H..
.....................................
............H..............H.........
.....................................
....................H..............H.
.....................................
.....................................
..H........H............H............
................H....................
..........................H..........
.....................................
.....................................
.....................................
.....................................
...........................H.........
.....................................
.....................................
.....................................
.....................................
...........H.........................
...........H.........................
...........H.........................
.....................................
.................H...................
.....................................
.......................H........H....
12 3
............
............
............
............
...H........
............
............
............
............
............
............
............
34 3
..................................
..................................
....................H.............
..................................
.................H................
..................................
..................................
..................H...............
..................................
..H..................H............
..................................
.....................H..H.........
..............H...................
............H.....................
..................................
..................................
..................................
..................................
............H.....................
..................................
..................................
..................................
..................................
..................................
..................H...........H...
...............................H..
....................H.............
..............................H...
..................................
..................................
........H.........................
......HH..........................
..................................
..................................
35 3
...................................
................................H..
...................................
...................................
.....................H.............
................................H..
...................................
...................................
.........H..........H..........H...
....................H..............
...................................
...................................
...................................
..............H....................
...................................
...................................
...................................
...................................
.....................H.............
.................H.................
...................................
...................................
...................................
...................................
...................................
.........H.........................
.............................H.....
......................H............
...................................
........H..........................
...................................
.....................H.............
...................................
...................................
...................................
8 3
........
.......H
........
........
........
......H.
.....H..
.H......
45 3
.........................H...................
.............................................
.....................H.......................
................H............................
.............................................
..........................H.........H........
...............H.............................
.............................................
.............................................
..H..........................................
..................................H..........
.........................H...................
..........................................H..
...........H.............H...H...............
.........................................H...
.............................................
.........................H...................
.............................................
.............................H...............
.............................................
.............................................
.............................................
.............................................
....................H........................
.............................................
.............................................
.............................................
........H....................H...............
.............................................
.......................H.....................
...........H.................................
.............................................
.............................................
.............................................
.............................................
..H..........................................
.............................................
.............................................
.......................................H.....
.............................................
.............................................
.............................................
.............................................
..........H..................................
.....................H...............H.......
3 3
...
H..
.H.
32 3
................................
................................
................................
..................H.............
................................
................................
................................
................................
..................H.............
................................
................................
................................
................................
.....H..........................
................................
..........H.....................
................................
................................
............................H...
................................
...............................H
................................
................................
................................
................................
................................
.............H...H........H.....
................................
............H...................
...............................H
................................
................................
5 3
.H...
.....
.....
..H..
.....
22 3
..........H...........
......................
.H....................
......................
................H.....
......................
.....H.......H...H....
......................
...H..................
......................
......................
......................
......................
......................
......................
......................
....H..........H......
......................
........H.H...........
......................
..............H.......
......................
34 3
..................................
...HH.............................
.................H................
..................................
...H..............................
..................................
.H................................
........H..H......................
..................................
.......H..........................
...............H..................
......................H...........
..................................
..................................
..................................
..................................
..................................
................................H.
..................................
..................................
..........H....H..................
..........................H.......
..................................
.........................H........
......H...........................
...H..............................
..................................
............................H.....
..................................
..................................
.................................H
..................................
..................................
.....................H............
32 3
................................
................................
.....H..........................
..............H.................
................................
................................
.H..............................
................................
....H...........................
................................
......................H.........
................................
............................H...
................................
................................
.............H..................
................................
................................
................................
................................
...................H......H.....
................................
................................
....................H...........
................................
................................
................................
........................H.......
............H...................
................................
.....................H..........
................................
6 3
....H.
H.....
H.H...
......
.H....
......
14 3
..............
...H..........
..............
..............
..............
............H.
..............
...........H..
..............
...H..........
H.............
.............H
.....H........
..............
48 3
................................................
................................................
.....H..........................................
...................................H............
......H...............H.........................
................................................
.............................H..................
.....H.......................H...H..............
................................................
................................................
................................................
.............H..................................
...H............................................
................................................
................................................
................................................
................................................
................................................
................................................
................................................
................................................
.....H..........................................
................................................
.................................H..............
.........H......................................
............................H...................
................................................
................................H...............
................................................
......H.........................................
................................................
................................................
................................................
......................H.........................
................................................
...............H................................
................................................
................................................
............................H...................
................................................
................................................
................................................
................................................
..........................H.....................
................................................
................................................
..........................H.....................
................................................
50 3
..................................................
..................................................
..................................................
..................................................
..................................................
.........................................H........
..................................................
.................................H................
..................................................
..................................................
..................................................
..................................................
.......................................H..........
......................................H........H..
........................................H.........
.................H......................H.........
..................................................
H.................................................
.....H..........H.................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
...H..............................................
..................................................
..................................................
...........................H......................
.H................................................
..................................................
......H.............H.......H.....................
..................................................
.........H........................................
.......................................H..........
................................H.................
..................................................
..............................H...................
.............................H....................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
..................................................
....................................H.............
45 3
.............................................
.............................................
...........................................H.
.............................................
.............................................
.............................................
...................H.........................
.............................................
..................H..........................
.............................................
..........H..................................
.....H.......................................
.............................................
.............................H...............
.........................H...H...............
.............................................
.............................................
.............................................
.............................................
.............................................
.............................H...............
............................H................
.............................................
.............................................
.............................................
.............H...............................
..............H..............................
.............................................
.............................................
.......H.....................................
................H.......H....................
........H....................................
.............................................
.............................................
.............................................
..H....H......H........................H....H
..............H............................H.
.............................................
.....H.......................................
.............................................
.............................................
.............................................
.............................................
.............................................
........................H....................
34 3
..................................
..........................H.......
.................H................
..................................
...............H...H..............
..................................
..............................H...
..................................
...............................H..
............H.....................
..................................
..................................
.........H........................
..................................
..................................
..................................
.........H..H.....................
..................................
..................................
......................H...........
.............................H....
..................................
...........................H......
....................H.............
.........................H........
..................................
..................................
.......H..........................
..................................
...............................H..
..................................
..................................
..................................
...H..............................
14 3
..............
.H............
...........H..
....H........H
..............
..............
..............
..............
..............
..........H...
..............
.H............
H.............
..............
18 3
..................
..............H...
..........H.......
..................
..................
...........H......
..................
.................H
..................
.........H........
..................
........H.........
..................
..................
..........H....H..
.................H
.H................
..................
4 3
....
....
..H.
....
