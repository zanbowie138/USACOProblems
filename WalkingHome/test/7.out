99
89
165
474
109
4
1802
501
386
1805
29
200
16
439
958
73
1036
1120
54
19
1477
63
11
38
374
892
935
223
170
625
200
1230
1449
42
649
2
697
8
333
204
1176
8
76
2862
1541
651
1126
162
165
8
